solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1191,9 +1192,11 @@ protected SolrCore registerCore(CoreDescriptor cd, SolrCore core, boolean regist
      * set both the name of the descriptor and the name of the
      * core, since the descriptors name is used for persisting.
      */
-
     core.setName(cd.getName());
-
+    
+    // Check and evict any pre-existing concurrency metadata for cores belonging to shared replicas
+    evictSharedCoreMetadata(cd);
```

Comment:
This call is not exclusive to cloud mode(zkSys.getZkController() != null). Inside evictShareCoreMetadata, getSharedStoreManager() can be null. Need zkSys.getZkController() != null protection here or inside evictShareCoreMetadata. I would put it inside the method.